sweep/sweeper.go

```diff
@@ -603,21 +670,28 @@ func (s *UtxoSweeper) getInputLists(currentHeight int32,
 // sweep takes a set of preselected inputs, creates a sweep tx and publishes the
 // tx. The output address is only marked as used if the publish succeeds.
 func (s *UtxoSweeper) sweep(inputs inputSet,
-	satPerKW lnwallet.SatPerKWeight, currentHeight int32) error {
+	feePreference FeePreference, currentHeight int32) error {
 
-	var err error
-
-	// Generate output script if no unused script available.
-	if s.currentOutputScript == nil {
-		s.currentOutputScript, err = s.cfg.GenSweepScript()
+	// Generate output script if there isn't an unused script available for
+	// the given fee preference.
+	if s.scriptsByFeePref[feePreference] == nil {
+		pkScript, err := s.cfg.GenSweepScript()
 		if err != nil {
 			return fmt.Errorf("gen sweep script: %v", err)
 		}
+		s.scriptsByFeePref[feePreference] = pkScript
+	}
+
+	// Retrieve the fee rate we'll attempt to use for the sweep transaction
+	// based of the given fee preference.
+	satPerKW, err := DetermineFeePerKw(s.cfg.FeeEstimator, feePreference)
```

Comment:
Is it necessary to call this again? We already made the call when we created the input set.

Reply:
Not really, removed.